operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,191 @@
+#include "draw_bounding_box.hpp"
+#include <algorithm>
+#include <gsl/narrow>
+#include <gsl/span>
+#include <vector>
+
+namespace ort_extensions {
+namespace {
+constexpr uint8_t KColorMap[10][3] = {
+    {255, 0, 0}, {0, 255, 0}, {0, 0, 255}, {255, 255, 0}, {255, 0, 255}, {128, 0, 0}, {0, 128, 0}, {0, 0, 128}, {128, 128, 128}, {0, 0, 0}};
+
+// Used to store image data, width, height, and number of channels
+struct ImageView {
+  uint8_t* data;
+  int64_t height;
+  int64_t width;
+  int64_t channels;
+};
+
+// To represent Boxes, [num of boxes, box_info]
+class RankTwoVector {
+ public:
+  RankTwoVector(const std::vector<int64_t>& shape, const float* data) : shape_(shape), data_(data) {
+  }
+
+  [[nodiscard]] const float* GetFirstDimDataPtr(size_t index) const {
+    return data_ + index * shape_.back();
+  }
+
+  [[nodiscard]] int64_t ShapeAtDim(int64_t dim) const {
+    return shape_[dim];
+  }
+
+ private:
+  gsl::span<const int64_t> shape_;
+  const float* data_;
+};
+
+// Draw a line on the image
+void draw_line_horizon(ImageView& image, int64_t point_start, int64_t line_length, const uint8_t* color,
+                       int64_t thickness) {
+  auto stride = image.width * image.channels;
+  for (int64_t i = 0; i < thickness; ++i) {
+    auto* end = image.data + point_start + i * stride + line_length * image.channels;
+    for (auto* start = image.data + point_start + i * stride; start < end; start += image.channels) {
+      std::copy_n(color, image.channels, start);
+    }
+  }
+}
+
+void draw_line_vertical(ImageView& image, int64_t point_offset, int64_t line_length, const uint8_t* color,
+                        int64_t thickness) {
+  auto* point_start = image.data + point_offset;
+  auto* point_end = point_start + line_length * image.width * image.channels;
+  for (; point_start < point_end; point_start += image.width * image.channels) {
+    for (int64_t i = 0; i < thickness; ++i) {
+      // no boundary check
+      std::copy_n(color, image.channels, point_start + i * image.channels);
+    }
+  }
+}
+
+// Draw a box on the image with given thickness and color
+void draw_box(ImageView& image, const float* box,
+              const uint8_t* color, int64_t thickness) {
+  // -------(1)------
+  // |              |
+  //(2)            (4)
+  // |              |
+  // -------(3)------
+  int64_t x_start = static_cast<int64_t>(box[0]);
+  int64_t y_start = static_cast<int64_t>(box[1]);
+  int64_t x_end = static_cast<int64_t>(box[2]);
+  int64_t y_end = static_cast<int64_t>(box[3]);
+
+  auto half_thickness = (thickness - 1) / 2;
+  x_start -= half_thickness;
+  y_start -= half_thickness;
+  x_end += half_thickness;
+  y_end += half_thickness;
+
+  x_start = std::max(x_start, 0L);
+  y_start = std::max(y_start, 0L);
+  x_end = std::min(x_end, image.height - 1);
+  y_end = std::min(y_end, image.width - 1);
+
+  if (x_end - x_start < thickness || y_end - y_start < thickness) {
+    // "invalid thickness, The box is too small to draw.";
```

Comment:
Silent failures are typically bad and hard to debug. Should this throw so the user can fix how the operator is being used?

Reply:
Personally, I would we can draw as much as possible boxes and one box failure shouldn't hold down the whole procedure.
But I agree with that silent failures is hard to debug.

Reply:
Can we just set the thickness to x_end - s_start or y_end - y_start? The line is so thick it will fill the box, so rather than overfilling the box, just fill it. Hopefully that equates to we draw the box, there's no error, and because the thickness or box is insane the user just gets a solid block.

Reply:
Yes. It's a acceptable compromise